source/server/exec.cpp

```diff
@@ -130,17 +130,19 @@ void Server::execNICK(Client& client, const std::string line)
 
     if (client.getNickname().empty())
     {
+        client.setNickname(name);
         std::cout << "Nick " << name << " has been added.\n";
+
     }
     else
     {
-        Utils::writeTo("Nickname " + name + " set.\n", client.getId());
+        client.setNickname(name);
+        replyWelcome(client);
         std::cout << "Nickname " << name << " set.\n";
 
     }
```

Comment:
O replyWelcome não deveria ser chamado se o nick for definido pela 1ª vez?
O else statement é um pouco excusado, podes fazer só:
```
if (client.getNickname().empty())
{
    replyWelcome(name);
    std::cout << ...;
}
client.setNickname(name);
client.setNickActive();
```